### DISCUSSION TITLE
Unknown interaction

### DISCUSSION THREAD

**Discussion by ghost** [discordjs/discord.js#9630] [Q&A] [answered]
### Which package is this bug report for?

discord.js

### Issue description

Hi. I'm using discord.js version 14.11.0, and after my first line of code, the interaction no longer works. I don't know if it's either me or discord, but I would like to know. 

### Code sample

```typescript
purge.js(just my example)
const { PermissionsBitField } = require("discord.js")

module.exports.run = async (inter) => {
    let amount = inter.options.getNumber("amount")
    await inter.deferReply()
    if(amount <= 100){
        inter.channel.bulkDelete(amount, true)
        inter.reply(`successfully purged ${amount} messages`)
    }
}

module.exports.help = {
    name: 'purge',
    permissions: [PermissionsBitField.Flags.ManageMessages]
}
interactionCreate.js
const Client = require("../index").Client
Client.on('interactionCreate', async inter => {
    if(inter.isCommand()){
        let slashCmds = Client.slashCmds.get(inter.commandName)
        if(inter.member.permissions.has(slashCmds.help.permissions)){
            if(slashCmds) slashCmds.run(inter)
        }
    }
})
index.js event loader.
fs.readdirSync('./events/').forEach(dir => {
  var jsFiles = fs.readdirSync('./events/').filter(f => f.split(".").pop() === "js");
  if (jsFiles.length <= 0) return console.log("[EVENT HANDLER] - Can't find any events to load.")

    let check = false
    jsFiles.forEach(event => {
        const eventGEt = require(`./events/${event}`)

        try{
            Client.events.set(eventGEt.name, eventGEt)
            if(check == false) {
                console.log(`[EVENT HANDELER] - File ${event} was loaded`)
                check = true
            }
        } catch(error){
            return console.log(error)
        }
    });

});
```

### Versions

- discord.js: 14.11.0
- NodeJS: v18.16.0

### Issue priority

Medium (should be fixed soon)

### Which partials do you have configured?

No Partials

### Which gateway intents are you subscribing to?

Guilds, GuildMembers, GuildModeration, GuildEmojisAndStickers, GuildIntegrations, GuildWebhooks, GuildInvites, GuildVoiceStates, GuildPresences, GuildMessages, GuildMessageReactions, GuildMessageTyping, DirectMessages, DirectMessageReactions, DirectMessageTyping, MessageContent, GuildScheduledEvents, AutoModerationConfiguration, AutoModerationExecution

### I have tested this issue on a development release

_Yes, I have_

**Comment by Vaibhav-Naik-26** [comment, marked as answer]
Try this
```js
// interactionCreate.js
const Client = require("../index").Client;

Client.on('interactionCreate', async (inter) => {
  if (inter.isCommand()) { // OR try `inter.isChatInputCommand()`
    const slashCmds = Client.slashCmds.get(inter.commandName);
    if (!slashCmds) return;

    if (inter.member.permissions.has(slashCmds.help.permissions))
      if (slashCmds) slashCmds.run(inter);
  };
});
```
```js
// purge.js(just my example)
const { PermissionsBitField } = require("discord.js")

module.exports.run = async (inter) => {
    const amount = inter.options.getNumber("amount");
    await inter.deferReply()
    if(amount <= 100){
        await inter.channel.bulkDelete(amount, true);
        await inter.followUp(`successfully purged ${amount} messages`);
    };
};
```

**Reply by ghost** [reply]
It worked. Thanks alot!

**Reply by Vaibhav-Naik-26** [reply]
> It worked. Thanks alot!

Mark this discussion as Answered

**Reply by Jiralite** [reply]
A good answer actually explains the issue and explains the solution, not spoon-feeds.

**Reply by Vaibhav-Naik-26** [reply]
> A good answer actually explains the issue and explains the solution, not spoon-feeds.

I agree to you and mine is also the same thoughts, I didn't do it to increase my reputation, i just told him to do so that no one else waste their time answering it.
If you think I did wrong, then I am very sorry, I will remember this now on-wards

And I just now saw that you closed this so... Thank You :)
